src/Format2.ps1

```diff
@@ -0,0 +1,194 @@
+        $v = ([PSObject]$Value.$p)
+        $f = Format-Nicely2 -Value $v -Pretty:$Pretty
+        "$p=$f"
+    }
+
+    if (0 -eq $Property.Length ) {
+        $o = "$valueType{}"
+    }
+    elseif ($Pretty) {
+        $o = "$valueType{`n    $($items -join ";`n    ");`n}"
+    }
+    else {
+        $o = "$valueType{$($items -join '; ')}"
+    }
+
+    $o
+}
+
+function Format-String2 ($Value) {
+    if ('' -eq $Value) {
+        return '<empty>'
+    }
+
+    "'$Value'"
+}
+
+function Format-Null2 {
+    '$null'
+}
+
+function Format-Boolean2 ($Value) {
+    '$' + $Value.ToString().ToLower()
+}
+
+function Format-ScriptBlock2 ($Value) {
+    '{' + $Value + '}'
+}
+
+function Format-Number2 ($Value) {
+    [string]$Value
+}
+
+function Format-Hashtable2 ($Value) {
+    $head = '@{'
+    $tail = '}'
+
+    $entries = foreach ($v in $Value.Keys | & $SafeCommands['Sort-Object']) {
+        $formattedValue = Format-Nicely2 $Value.$v
+        "$v=$formattedValue"
+    }
+
+    $head + ( $entries -join '; ') + $tail
+}
+
+function Format-Dictionary2 ($Value) {
+    $head = 'Dictionary{'
+    $tail = '}'
+
+    $entries = foreach ($v in $Value.Keys | & $SafeCommands['Sort-Object'] ) {
+        $formattedValue = Format-Nicely2 $Value.$v
+        "$v=$formattedValue"
+    }
+
+    $head + ( $entries -join '; ') + $tail
+}
+
+function Format-Nicely2 ($Value, [switch]$Pretty) {
+    if ($null -eq $Value) {
+        return Format-Null2 -Value $Value
+    }
+
+    if ($Value -is [bool]) {
+        return Format-Boolean2 -Value $Value
+    }
+
+    if ($Value -is [string]) {
+        return Format-String2 -Value $Value
+    }
+
+    if ($value -is [type]) {
+        return Format-Type2 -Value $Value
+    }
+
+    if (Is-DecimalNumber -Value $Value) {
+        return Format-Number2 -Value $Value
+    }
+
+    if (Is-ScriptBlock -Value $Value) {
+        return Format-ScriptBlock2 -Value $Value
+    }
+
+    if (Is-Value -Value $Value) {
+        return $Value
+    }
+
+    if (Is-Hashtable -Value $Value) {
+        return Format-Hashtable2 -Value $Value
+    }
+
+    if (Is-Dictionary -Value $Value) {
+        return Format-Dictionary2 -Value $Value
+    }
+
+    if ((Is-DataTable -Value $Value) -or (Is-DataRow -Value $Value)) {
+        return Format-DataTable2 -Value $Value -Pretty:$Pretty
+    }
+
+    if (Is-Collection -Value $Value) {
+        return Format-Collection2 -Value $Value -Pretty:$Pretty
+    }
+
+    Format-Object2 -Value $Value -Property (Get-DisplayProperty2 $Value.GetType()) -Pretty:$Pretty
+}
+
+function Get-DisplayProperty2 ([Type]$Type) {
+    # rename to Get-DisplayProperty?
+
+    <# some objects are simply too big to show all of their properties,
+    so we can create a list of properties to show from an object
+    maybe the default info from Get-FormatData could be utilized here somehow
+    so we show only stuff that would normally show in format-table view
+    leveraging the work PS team already did #>
```

Comment:
Experimented a little with this. Example:
```powershell
# root of Format2.ps1
$script:defaultDisplayPropertyCache = [System.Collections.Generic.Dictionary[type,string[]]]::new()

function Get-DisplayProperty2 ($Object) {
    $propertyMap = @{
        [System.Diagnostics.Process] = 'Id', 'Name'
    }

    $type = $Object.GetType()

    $properties = $propertyMap[$type]
    if ($null -ne $properties) { return $properties }

    # Fallback to DefaultDisplayPropertySet if defined for type
    if ($script:defaultDisplayPropertyCache.TryGetValue($type,[ref]$properties)) { return $properties }

    # Alterantive to enable inheritance: Loop $Object.PSTypeNames with (Get-TypeData $typename).DefaultDisplayPropertySet.ReferencedProperties
    # Would be slower and not sure if any types work this way
    if (($standardMembers = $Object.psobject.Members['PSStandardMembers']) -and
        ($defaultDisplayPropSet = $standardMembers.psobject.Members['DefaultDisplayPropertySet'])) {
        $properties = $defaultDisplayPropSet.ReferencedPropertyNames -as [string[]]
    }
    # Always cache to skip repeating PSStandardMembers check
    $script:defaultDisplayPropertyCache.Add($type, $properties)
    $properties
```

Thoughts:
- Types like `DirectoryInfo` do not have standard member set, so needs to go in the `$propertyMap` to avoid infinite loop due to `Root`-property.
- Is it a problem that `Get-DisplayProperty2` could exclude properties that were different? Should be visible with `-Verbose` but still?

Reply:
moved: https://github.com/pester/Pester/issues/2474